### ISSUE TITLE
MVP architecture Implementation 

### ISSUE THREAD

**Issue by therajanmaurya** [BenGoBlue05/UdacityAlumni#55]
### Proposal for implementing the MVP architecture by [Google](https://github.com/googlesamples/android-architecture/tree/todo-mvp-rxjava)  or [Ribot](https://github.com/ribot/android-boilerplate) 
We can follow any if them, I can take responsibility for this.

I am suggesting the MVP architecture then It will become easy to write the unit test and Integration test and without any architecture, It's very hard to write the test cases.

What do you think @BenGoBlue05 @PPartisan @chashmeetsingh

**Comment by PPartisan** [comment]
I agree MVP would make it much simpler to write tests, but my concern is that formally adopting it would put off new contributors. MVP isn't something covered in the AND, so there's a good chance many current students and alumni wont have experience with it.

**Comment by therajanmaurya** [comment]
Yeah, you are right, MVP was not covered in AND but It is good if the developer knows the technology. 
During the implementation of MVP, we can write some docs and blogs and can also refer already written.

So what you think.

**Comment by abhishek-verma** [comment]
I am familiar with MVP, I'd like to contribute.

**Comment by imGurpreetSK** [comment]
If still relevant, we can create a new branch named UdacityAlumni-MVP and push the MVP code there. Will this work?

**Comment by n8ebel** [comment]
I think having separate branches to demonstrate different architectures is a great idea.  Could be a wonderful "next step" for new grads that want to start exploring Android beyond the scope of the Android Nanodegree program

**Comment by knightcube** [comment]
Can i work on this?
